src/jit/utils.cpp

```diff
@@ -1954,6 +1954,42 @@ float FloatingPointUtils::round(float x)
     return _copysignf(flrTempVal, x);
 }
 
+bool FloatingPointUtils::isNormal(double x)
+{
+    int64_t bits = *(int64_t*)&x;
+    bits &= 0x7FFFFFFFFFFFFFFF;
+    return (bits < 0x7FF0000000000000) && (bits != 0) && ((bits & 0x7FF0000000000000) != 0);
+}
+
+bool FloatingPointUtils::isNormal(float x)
+{
+    int32_t bits = *(int32_t*)&x;
+    bits &= 0x7FFFFFFF;
+    return (bits < 0x7F800000) && (bits != 0) && ((bits & 0x7F800000) != 0);
+}
+
+bool FloatingPointUtils::hasPreciseReciprocal(double x)
+{
+    if (!isNormal(x))
+        return false;
+
+    uint64_t i = *(uint64_t*)&x;
+    uint64_t exponent = (i >> 52) && 0x7FFul;
+    uint64_t mantissa = i & 0xFFFFFFFFFFFFFul;
+    return mantissa == 0 && exponent != 0 && exponent != 1023;
```

Comment:
nit: Might be worth a comment explaining how this works (I believe @CarolEidt, generally prefers function headers to be added as well).

Namely, it's worth calling out that this is checking for powers of 2 and that we don't handle denormals since those may not be well-defined on some platforms (such as if the user modified the floating-point environment via a P/Invoke) and that this includes returning false both when the input is denormal and when the reciprocal is denormal (the exponent is zero).

Reply:
> I believe @CarolEidt, generally prefers function headers to be added as well

It's actually in the JIT coding conventions here: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#9.4

Reply:
Oh, forgot about this PR :( Added comments to `hasPreciseReciprocal` let me know if you also need comments for trivial `isNormal`.

Reply:
From the doc it's not clear for me should I write comments in .cpp or .h